src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,50 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProducer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProducer;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Track;
+import core.mate.academy.model.TrackProducer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (Bulldozer.class.equals(type)) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            return new ArrayList<>(bulldozerProducer.get());
+        }
+        if (Excavator.class.equals(type)) {
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+            return new ArrayList<>(excavatorProducer.get());
+        }
+        if (Track.class.equals(type)) {
+            TrackProducer trackProducer = new TrackProducer();
+            return new ArrayList<>(trackProducer.get());
+        }
+        return new ArrayList<>(0);
```

Comment:
```suggestion
        return new ArrayList<>();
```
for future:
you can use Collection.emptyList() instead of new ArrayList<>()